rhio/examples/topics.rs

```diff
@@ -0,0 +1,69 @@
+use std::time::Duration;
+
+use anyhow::Result;
+use p2panda_core::PrivateKey;
+use rhio::config::Config;
+use rhio::messages::{Message, MessageContext};
+use rhio::node::Node;
+use rhio::topic_id::TopicId;
+
+/// The only message type in our chat app
+type ChatMessage = String;
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    let private_key = PrivateKey::new();
+    let mut config = Config::default();
+
+    // Add a topic to the node configuration
+    let chat_topic_id = TopicId::from_str("my_chat");
+    config.topics.push(chat_topic_id);
+
+    // Spawn the node
+    let mut node: Node<ChatMessage> = Node::spawn(config.clone(), private_key.clone()).await?;
+
+    println!("Peer Id: {}", private_key.public_key().to_hex());
+
+    println!("joining gossip overlay ..");
+    let _ = node.ready().await;
+    println!("gossip overlay joined!");
+
+    // Get channels for sending and receiving messages on the chat topic
+    let (chat_tx, mut chat_rx) = node.topic(chat_topic_id).await?;
```

Comment:
> Is it possible to join topics here which have not been mentioned in the config?

No, this isn't possible at the moment (see comment below).

Reply:
> > Is it possible to join topics here which have not been mentioned in the config?
> 
> No, this isn't possible at the moment (see comment below).

It would be surprising to see it fail here. I find this confusing, but I'm also fine to leave it as it'll anyhow be changed in the future (see other comment).